File: web/types/crisis.ts
export type CrisisType =
  | "flood"
  | "wildfire"
  | "health_emergency"
  | "storm"
  | "other";

export type ResponseState =
  | "needs_identified"
  | "ping_sent"
  | "response_confirmed"
  | "gap_flagged";

export interface CrisisMapPoint {
  latitude: number;
  longitude: number;
}

export interface CommunityProfile {
  id: string;
  crisis_id: string;
  location: string;
  vulnerability_score: number;
  elderly_pct: number;
  spanish_speaking_pct: number;
  low_income_pct: number;
  mobility_limited_pct: number;
  top_needs: string[];
}

export interface Organization {
  id: string;
  name: string;
  services: string[];
  languages: string[];
  coverage_areas: string[];
  capacity: string;
}

export interface ResponseTracking {
  id: string;
  crisis_id: string;
  org_id: string;
  needs_covered: string[];
  status: ResponseState;
  pinged_at: string | null;
  confirmed_at: string | null;
  organization: Organization;
}

export interface GapAlert {
  id: string;
  crisis_id: string;
  unmet_needs: string[];
  alert_message: string;
  escalation_recommendation: string;
  fired_at: string;
}

export interface CrisisDetail {
  id: string;
  alert_text: string;
  crisis_type: CrisisType;
  severity: string;
  location: string;
  affected_population: number;
  risk_flags: string[];
  created_at: string;
  response_state: ResponseState;
  coordinates: CrisisMapPoint;
  community_profile: CommunityProfile | null;
  responses: ResponseTracking[];
  gap_alerts: GapAlert[];
}

export interface CrisisListResponse {
  total: number;
  items: CrisisDetail[];
}

export interface GapDetectionResponse {
  gap_detected: boolean;
  alert_status?: string;
  unmet_needs: string[];
  alert_message: string;
  escalation_recommendation: string;
  escalation_org?: string;
}

export interface SignalSource {
  raw_text: string;
  timestamp: string;
}

export interface SignalIntelligence {
  source_count?: number;
  signal_confidence?: "high" | "medium" | "low" | string;
  severity_escalation?: boolean;
  escalation_reason?: string;
  unified_alert_text?: string;
  location?: string;
  severity?: string;
  crisis_type?: CrisisType | string;
  vulnerability_score?: number;
  fema?: SignalSource;
  weather?: SignalSource;
  weather_gov?: SignalSource;
  news?: SignalSource;
  social?: SignalSource;
  FEMA?: SignalSource;
  "Weather.gov"?: SignalSource;
  News?: SignalSource;
  Social?: SignalSource;
  fema_signal?: SignalSource;
  weather_signal?: SignalSource;
  news_signal?: SignalSource;
  social_signal?: SignalSource;
}

File: backend/schemas/crisis.py
from datetime import datetime

from pydantic import BaseModel


class CrisisMapPoint(BaseModel):
    latitude: float
    longitude: float


class CommunityProfileResponse(BaseModel):
    id: str
    crisis_id: str
    location: str
    vulnerability_score: int
    elderly_pct: float
    spanish_speaking_pct: float
    low_income_pct: float
    mobility_limited_pct: float
    top_needs: list[str]


class OrganizationResponse(BaseModel):
    id: str
    name: str
    services: list[str]
    languages: list[str]
    coverage_areas: list[str]
    capacity: str


class ResponseTrackingResponse(BaseModel):
    id: str
    crisis_id: str
    org_id: str
    needs_covered: list[str]
    status: str
    pinged_at: datetime | None
    confirmed_at: datetime | None
    organization: OrganizationResponse


class GapAlertResponse(BaseModel):
    id: str
    crisis_id: str
    unmet_needs: list[str]
    alert_message: str
    escalation_recommendation: str
    fired_at: datetime


class CrisisTriggerRequest(BaseModel):
    location: str


class CrisisSummaryResponse(BaseModel):
    id: str
    alert_text: str
    crisis_type: str
    severity: str
    location: str
    affected_population: int
    risk_flags: list[str]
    created_at: datetime
    response_state: str
    coordinates: CrisisMapPoint


class CrisisDetailResponse(CrisisSummaryResponse):
    community_profile: CommunityProfileResponse | None
    responses: list[ResponseTrackingResponse]
    gap_alerts: list[GapAlertResponse]


class CrisisListResponse(BaseModel):
    total: int
    items: list[CrisisDetailResponse]


class ResponseConfirmationRequest(BaseModel):
    crisis_id: str
    org_id: str
    needs_covered: list[str]


class SimulateElapsedRequest(BaseModel):
    crisis_id: str


class GapStatusResponse(BaseModel):
    gap_detected: bool
    alert_status: str
    alert_message: str
    escalation_recommendation: str
    escalation_org: str
    unmet_needs: list[str]
